Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -260,109 +278,122 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
         }
 
         // <inheritdoc />
-        public async Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl)
+        public async Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl, ITracer tracer)
         {
-            ITracer tracer = _traceFactory.GetTracer();
+            var installationLock = InstallationLock.CreateLock(_environment.SiteExtensionSettingsPath, id);
+            // hold on to lock till action complete (success or fail)
+            return await installationLock.Hold<SiteExtensionInfo>(async () =>
```

Comment:
removed "Hold" method, use LockOperationAsync with Timespan.zero as timeout from LockExtensions.cs

if not able to aquire lock, LockOperationAsync will throw exception